src/document/doc-builders.js

```diff
@@ -56,7 +56,7 @@ function indent(contents) {
 }
 
 /**
- * @param {number} n
+ * @param {number | string} n
```

Comment:
How about this:

```suggestion
 * @param {any} n
```

This is assuming that we would plan to do `git grep` for JSDoc `any` types and make them more specific at some point in the future.

Reply:
I send a PR rewrite this part https://github.com/prettier/prettier/pull/8931/files

Do you think you can make it specific in this PR?

Reply:
I think you mean adding specific types to PR #8931, right?

I would like to keep this PR simple, since it is to enable typecheck on several (4) languages.

Maybe we should keep this PR in draft until we fix the types in `align` and `makeAlign`?

Reply:
No, I mean this PR, #8931 only mean to make logic cleaner  for you.

Reply:
> I would like to keep this PR simple, since it is to enable typecheck on several (4) languages.

But this PR requires a fix for `align` function, and what I did is not good enough.

Reply:
> Maybe we should keep this PR in draft until we fix the types in align and makeAlign?

Let's only take care of `align`.

Reply:
After reading and experimenting with some of the code, I think this is what we need:

```suggestion
 * @param {number | string | Doc} n
```

I think this is the one place that the Doc type option is needed for: https://github.com/prettier/prettier/blob/9dad95b35f935edce4c3d6cfa45c79a0b9c82b9f/src/document/doc-builders.js#L104-L106

I am now pushing changes to apply this suggestion and get rid of that old `@ts-ignore` comment.

Reply:
What  about  `if(!n)`, this  contains many  possibility. could be  `undefined` `null` `false` `NaN`.

Reply:
Let's ignore it for now, I've add `TODO` there.